modules/swagger-core/src/main/java/io/swagger/v3/core/util/ReflectionUtils.java

```diff
@@ -51,39 +51,54 @@ public static Type typeFromString(String type) {
     }
 
     /**
-     * Checks if the method methodToFind is the overridden method from the superclass or superinterface.
+     * Checks if the method methodToFind is the overridden method from the superclass or superinterface. In most cases getMethods should be called instead.
      *
      * @param methodToFind is method to check
      * @param cls          is method class
      * @return true if the method is overridden method
      */
     public static boolean isOverriddenMethod(Method methodToFind, Class<?> cls) {
-        Set<Class<?>> superClasses = new HashSet<>();
-        for (Class c : cls.getInterfaces()) {
-            superClasses.add(c);
-        }
-
-        if (cls.getSuperclass() != null) {
-            superClasses.add(cls.getSuperclass());
-        }
+      return !getMethods(cls).contains(methodToFind);
+    }
 
-        for (Class<?> superClass : superClasses) {
-            if (superClass != null && !(superClass.equals(Object.class))) {
-                for (Method method : superClass.getMethods()) {
-                    if (method.getName().equals(methodToFind.getName()) && method.getReturnType().isAssignableFrom(methodToFind.getReturnType())
-                            && Arrays.equals(method.getParameterTypes(), methodToFind.getParameterTypes()) && !Arrays.equals(method.getGenericParameterTypes(), methodToFind.getGenericParameterTypes())) {
-                        return true;
+    /**
+     * Returns the list of methods of the given class and its hierarchy but without overridden methods.
+     * It is similar to Class.getMethods but overridden ones are not listed.
+     *
+     * @param cls          is the class
+     * @return the list of methods but without overridden methods
+     */
```

Comment:
The Class.getMethods already scans over the hierarchy so, It is not needed to load the hierarchy.